src/Sulu/Bundle/MediaBundle/Media/Storage/S3Storage.php

```diff
@@ -35,19 +40,20 @@ public function __construct(FilesystemInterface $filesystem, int $segments)
             throw new \RuntimeException('This storage can only handle filesystems with "AwsS3Adapter".');
         }
 
-        /** @var AwsS3Adapter $adapter */
-        $adapter = $filesystem->getAdapter();
+        $this->adapter = $filesystem->getAdapter();
 
-        $this->endpoint = (string) $adapter->getClient()->getEndpoint();
-        $this->bucketName = $adapter->getBucket();
+        $this->endpoint = (string) $this->adapter->getClient()->getEndpoint();
+        $this->bucketName = $this->adapter->getBucket();
     }
 
     public function getPath(array $storageOptions): string
     {
         $segment = $this->getStorageOption($storageOptions, 'segment');
         $fileName = $this->getStorageOption($storageOptions, 'fileName');
 
-        return $this->endpoint . '/' . $this->bucketName . '/' . $segment . '/' . $fileName;
+        $path = $this->adapter->applyPathPrefix($segment . '/' . $fileName);
```

Comment:
Should we `ltrim` the `/` here or does this handle applyPathPrefix correctly?

Reply:
yes it handles it correctly - but it makes sens to trim it!